databricks/koalas/series.py

```diff
@@ -1452,6 +1452,138 @@ def clip(self, lower: Union[float, int] = None, upper: Union[float, int] = None)
+        ...                       [0, 1, 2, 0, 1, 2, 0, 1, 2]])
+        >>> s = ks.Series([45, 200, 1.2, 30, 250, 1.5, 320, 1, 0.3],
+        ...               index=midx)
+        >>> s
+        lama    speed      45.0
+                weight    200.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                weight      1.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop(labels='weight', level=1)
+        lama    speed      45.0
+                length      1.2
+        cow     speed      30.0
+                length      1.5
+        falcon  speed     320.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop(('lama', 'weight'))
+        lama    speed      45.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                weight      1.0
+                length      0.3
+        Name: 0, dtype: float64
+
+        >>> s.drop([('lama', 'speed'), ('falcon', 'weight')])
+        lama    weight    200.0
+                length      1.2
+        cow     speed      30.0
+                weight    250.0
+                length      1.5
+        falcon  speed     320.0
+                length      0.3
+        Name: 0, dtype: float64
+        """
+        index = labels
+        if index is not None:
+            if not isinstance(index, (str, list, tuple)):
+                raise ValueError("'index' type should be one of str, list, tuple")
+            if level is None:
+                level = 0
+            if level >= len(self._internal.index_scols):
+                raise ValueError("'level' should be less than the number of indexes")
+
+            if isinstance(index, str):
+                index = [index]
+            if isinstance(index, list):
+                if all((isinstance(idx, str) for idx in index)):
+                    sdf = self._internal.sdf.where(
+                        ~self._internal.index_scols[level].isin(index))
+                    return Series(self._internal.copy(sdf=sdf))
+
+                elif all((isinstance(idx, tuple) for idx in index)):
+                    df = self.to_frame().transpose().drop(index).transpose()
+                else:
+                    raise ValueError("If the given index is a list, it "
+                                     "should only contains names as strings, "
+                                     "or a list of tuples that contain "
+                                     "index names as strings")
+            if isinstance(index, tuple):
+                df = self.to_frame().transpose().drop(index).transpose()
+
+            internal = self._internal.copy(sdf=df._internal.sdf, scol=df._internal.scol)
+            return Series(internal)
```

Comment:
This will lost the anchor. We should do `_col(DataFrame( ... ))` with proper `_InternalFrame`.